src/System.IO.FileSystem/src/System/IO/FileStream.Win32.cs

```diff
@@ -1955,11 +1872,6 @@ private Task<int> TaskFromResultOrCache(int result)
 
         private void LockInternal(long position, long length)
         {
-            if (_fileHandle.IsClosed)
```

Comment:
Are you suggesting I add an additional Debug.Assert? I moved the check to the shared caller.